### PR TITLE
[5.1] KCRO-162: Improve security of the monster-ui forgot password feature

### PR DESCRIPTION
**Pull request by filipevarjao**
A vulnerability on the reset password endpoint allowing identify valid accounts by using the reset password requests, so now we always return success when requested: An email will be sent with details on how to reset your password. 
In addition, this request consumes tokens from token buckets.

### CODE REVIEW

File: src/modules/cb_user_auth.erl [hunk 1]
```diff
@@ -255,15 +255,16 @@ validate_action(Context, _Action) ->
 
 -spec validate(cb_context:context(), path_token()) -> cb_context:context().
 validate(Context, ?RECOVERY) ->
-    case cb_context:req_verb(Context) of
+    Context1 = consume_tokens(Context),
```
Comment (jamesaimonetti) [inline src/modules/cb_user_auth.erl:258]: Shouldn't you check the resp_status after consuming tokens, and end the request if the status is not 'success'?

Comment (filipevarjao) [reply src/modules/cb_user_auth.erl:258]: clever you

**Comment by jamesaimonetti** [comment]
Is this PR or https://github.com/2600hz/kazoo-crossbar/pull/488 wanted?

**Comment by filipevarjao** [comment]
Yes, I will update it tomorrow, we already merged the master

**Comment by manushiperera** [comment]
Hello @filipevarjao,
Any update on this?

**Comment by filipevarjao** [comment]
we already get it merged in a different pr.

**Comment by sanjeevie6353** [comment]
@filipevarjao can we delete the branch is this is not needed anymore?

**Comment by filipevarjao** [comment]
sure

**Pull request closed without merging**